### PR TITLE
Crypto module: make toHex routines use appendCodePointValues()

### PR DESCRIPTION
**Pull request by cassella**
This routine, provided by #24104, makes these string constructions much faster.

### CODE REVIEW

**Comment by cassella** [comment]
@mppf?

I had to make a copy of the private `convertNibble()` function.  I don't know where this should live as a shared routine.

`test/library/package/Crypto` passes.  These tests do cover some `toHex()` and `toHexString()` cases.

This change speeds up one of my aoc solutions from 1016 seconds down to 76.

**Comment by cassella** [comment]
Grepping through `modules` for `format`, I'm not seeing anything else that's obvious to convert.

File: modules/packages/Crypto.chpl [hunk 1]
```diff
@@ -1348,4 +1356,23 @@ proc bfEncrypt(plaintext: CryptoBuffer, key: CryptoBuffer, IV: CryptoBuffer, cip
 
     extern proc RAND_seed(const buf: c_ptr(void), num: c_int);
   }
+
+  /*
+   * Convert a nibble into a character in its hexadecimal representation.
+   * Copied from Bytes.chpl
```
Comment (mppf) [inline modules/packages/Crypto.chpl:1362]: Could you add

> TODO: put this in a shared location that both can use

**Review by mppf — APPROVED** [review]
I need to run testing on it before merging, but it's reviewed now

**Comment by mppf** [comment]
@cassella - please let me know once you've done the comment update I requested & then I'll be ready to merge this.

**Comment by cassella** [comment]
@mppf ping?

**Comment by mppf** [comment]
@cassella -- thanks. I'll plan to merge this on Monday.

**Pull request merged**